nautobot/docs/models/extras/dynamicgroup.md

```diff
@@ -1,7 +1,319 @@
+
+Piecing the rules together, we ended up with two filters and by wrapping them in their own set of parentheses we get our final generated filter:
+
+```no-highlight
+((first-child OR second-child) AND (NOT nested-child))
+```
+
+How does this work?
+
+| Start with   | Filter        | Operator | Weight | Group Count | Final Count |
+| ------------ | ------------- | -------- | ------ | ----------- | ----------- |
+| parent       | none          | none     | none   | 389         | 389         |
+| first-child  | site=ams01    | restrict | 10     | 11          | 11          |
+| second-child | site=ang01    | include  | 20     | 7           | 18          |
+| third-child  | none          | none     | 30     | 389         | n/a         |
+| nested-child | status=active | exclude  | 10     | 389         | 0           |
+
+### Assigning Child Groups
+
+TBD. Higher-weight fields will be ordered lower within a form. (The default weight is 100.)
+
+## Dynamic Groups and the REST API
+
+Dynamic Groups are fully supported by the API.
+
+### Creating a Dynamic Group
+
+Request:
+
+```json
+POST /api/extras/dynamic-groups/
+{
+    "name": "parent2",
+    "slug": "parent2",
+    "description": "I am the parent group with nested children.",
+    "content-type": "dcim.device"
+}
+```
+
+Response:
+
+```json
+200 OK
+Content-Type: application/json
+
+{
+    "id": "03df0328-337c-4eeb-9095-fa2764cb31e2",
+    "display": "parent",
+    "url": "http://localhost:6789/api/extras/dynamic-groups/03df0328-337c-4eeb-9095-fa2764cb31e2/",
+    "name": "parent",
+    "slug": "parent",
+    "description": "I am the parent group with nested children.",
+    "content_type": "dcim.device",
+    "filter": {},
+    "children": [],
+    "created": "2022-07-06",
+    "last_updated": "2022-07-06T20:17:04.305663Z",
+    "custom_fields": {},
+    "computed_fields": {},
+    "relationships": {}
+}
+```
+
+### Specifying Filter Conditions
+
+Dynamic Groups are fairly straightforward however it is important to note that the `filter` field is a JSON field and it must be able to be used as valid query parameters for filtering objects of the corresponding content type.
+
+!!! warning
+    _Changed in version 1.4.0_  <!-- markdownlint-disable-line MD036 -->
+
+    Prior to v1.4.0, any invalid field names that are not eligible for filtering objects will be discarded upon validation.
+
+    As of v1.4.0, [strict filtering is enabled by default](../../../configuration/optional-settings/#strict_filtering), which causes any invalid field names to result in a `ValidationError`.
+
+Any invalid field values for valid field names will also result in a `ValidationError`.
+
+Consider, for example, the following Dynamic Group:
```

Comment:
Is this an example of invalid field names or field values? Or is it a valid example? Not clear from context.